controllers/controllers/shared/index.go

```diff
@@ -0,0 +1,30 @@
+package shared
+
+import (
+	"context"
+
+	networkingv1alpha1 "code.cloudfoundry.org/cf-k8s-controllers/controllers/apis/networking/v1alpha1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+const DestinationAppName = "destinationAppName"
+
+func SetupIndexWithManager(mgr manager.Manager) error {
+
+	// Generate indexes for CFRoute on field spec.Destination.AppRef.Name for efficient querying.
+	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &networkingv1alpha1.CFRoute{}, DestinationAppName,
+		func(rawObj client.Object) []string {
+			route := rawObj.(*networkingv1alpha1.CFRoute)
+			var destinationAppNames []string
+			for _, destination := range route.Spec.Destinations {
+				destinationAppNames = append(destinationAppNames, destination.AppRef.Name)
+			}
+			return destinationAppNames
```

Comment:
Out of curiosity, how does this work with an array of strings?

Reply:
@tcdowney - in case of array, the field selector tries to match with at-least one value in the array. 
more info and example - https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client#FieldIndexer

Reply:
Awesome! Learned something new 😁